echidna/scripts/dump_scaled.py

```diff
@@ -9,43 +9,74 @@
-    parser.add_argument("-s", "--scale", default=None, type=float,
-                        help="specify scale factor"
-                        "e.g. 0.05 for 5 percent")
-    parser.add_argument("-f", "--file", default=None, type=str,
-                        help="specify path to hdf5 file")
-    args = parser.parse_args()
+_logger = utilities.start_logging()
+
+
+def main(args):
+    """ Scales and dumps spectra.
+
+    Args:
+      args (Namespace): Container for arguments. See: python dump_scaled.py -h
 
+    Raises:
+      IOError: If no file is given to scale
+      ValueError: If no scale factor is given
+      ValueError: If no parameter is given to scale.
+      ValueError: If destination directory does not exits.
+    """
     if not args.file:
         parser.print_help()
         raise IOError("No file given in command line to scale")
     if not args.scale:
         parser.print_help()
         raise ValueError("No scale factor given")
-    if not args.dimension:
+    if not args.par:
         parser.print_help()
-        raise ValueError("No dimension to scale given")
-
-    directory = args.file[:args.file.rfind("/")+1]  # strip filename
-    # strip directory and extension
-    filename = args.file[args.file.rfind("/")+1:args.file.rfind(".")]
+        raise ValueError("No parameter to scale given")
+    if args.dest:
+        if os.path.isdir(args.dest):
+            direcory = args.dest
+            if directory[-1] != "/":
+                directory += "/"
+        else:
+            raise ValueError("%s does not exist." % args.dest)
+    else:
+        directory = os.path.dirname(args.file) + "/"
+    filename = os.path.splitext(os.path.basename(args.file))[0]
     scaler = scale.Scale()
     scaler.set_scale_factor(args.scale)
     spectrum = store.load(args.file)
-    scaled_spectrum = scaler.scale(spectrum, args.dimension)
-    filename = directory + filename + "_scaled.hdf5"
+    scaled_spectrum = scaler.scale(spectrum, args.par)
+    str_sc = str(args.scale)
+    if str_sc[-2:] == '.0':
+        str_sc = str_sc[:-2]
+    str_sc.rstrip('0')
+    filename = directory + filename + "_" + str_sc + "sc.hdf5"
     store.dump(filename, scaled_spectrum)
+
+
+if __name__ == "__main__":
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-p", "--par", default=None, type=str,
+                        help="specify parameter to scale e.g. energy_mc"
+                        "e.g. 0.05 for 5 percent")
+    parser.add_argument("-s", "--scale", default=None, type=float,
+                        help="specify scale factor"
+                        "e.g. 0.05 for 5 percent")
+    parser.add_argument("-f", "--file", default=None, type=str,
+                        help="specify path to hdf5 file")
+    parser.add_argument("-d", "--dest", default=None, type=str,
+                        help="specify path to dump hdf5 file")
+    args = parser.parse_args()
+    main(args)
```

Comment:
It would be nice to have the end block in a `try...except` statement like I suggest in #146